packages/redis-cache/src/RedisStack.ts

```diff
@@ -0,0 +1,76 @@
+import { NEXT_CACHE_TAGS_HEADER } from 'next/dist/lib/constants'
+import {
+  createClient,
+  RedisClientType,
+  RedisClientOptions,
+  RedisModules,
+  RedisDefaultModules,
+  RedisFunctions,
+  RedisScripts,
+  SchemaFieldTypes
+} from 'redis'
+import type { CacheEntry } from '@dbbs/next-cache-handler-common'
+import { Cache } from '@dbbs/next-cache-handler-core'
+import { RedisAdapter, RedisJSON } from './types'
+import { CHUNK_LIMIT } from './constants'
+
+const INDEX_NAME = 'idx:revalidateByTag'
+const SEPARATOR = ','
+const REGEX_PUNCTUATION = /[^a-zA-Z0-9]/g
+
+export class RedisStack implements RedisAdapter {
+  client: RedisClientType<RedisDefaultModules & RedisModules, RedisFunctions, RedisScripts>
+
+  constructor(options: RedisClientOptions) {
+    this.client = createClient(options)
+
+    this.client.connect().then(() =>
+      this.client.ft._list().then((listOfIndexes) => {
+        if (listOfIndexes.includes(INDEX_NAME)) return
+        this.client.ft
+          .create(INDEX_NAME, { '$.tags': { type: SchemaFieldTypes.TAG, AS: 'tag', SEPARATOR } }, { ON: 'JSON' })
+          .then(() => Cache.logger.info('Index created successfully.'))
+          .catch((e) => {
+            Cache.logger.error('Could not create an index for revalidating by tag', e)
```

Comment:
we should fail build if we were not able to create index, because without index it won't work

Reply:
I've added re-throwing an error. Is it enough?